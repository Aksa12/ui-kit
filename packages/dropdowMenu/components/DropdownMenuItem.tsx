import * as React from "react";
import { cx } from "emotion";
import {
  menuListItemActive,
  menuListItemSelected,
  menuListItemSelectedActive
} from "../style";
import { padding, margin } from "../../shared/styles/styleUtils";

export interface DropdownMenuItemProps extends React.HTMLProps<HTMLDivElement> {
  children: React.ReactNode;
  disabled?: boolean;
  index: number;
  listLength: number;
  isActive?: boolean;
  isSelected?: boolean;
}

const DropdownMenuItem = (props: DropdownMenuItemProps) => {
  const { isActive, isSelected, index, listLength, disabled, ...other } = props;
  return (
    <div
      className={cx(padding("horiz"), padding("vert", "xs"), {
        [menuListItemActive]: isActive,
        [menuListItemSelected]: isSelected,
        [menuListItemSelectedActive]: isActive && isSelected,
        [margin("top", "xs")]: index === 0,
        [margin("bottom", "xs")]: index === listLength - 1
      })}
      {...other}
    />
  );
};

export default DropdownMenuItem;
