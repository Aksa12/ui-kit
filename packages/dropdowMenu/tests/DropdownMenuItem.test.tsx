import React from "react";
import { shallow } from "enzyme";
import * as emotion from "emotion";
import { createSerializer } from "jest-emotion";
import toJson from "enzyme-to-json";
import DropdownMenuItem from "../components/DropdownMenuItem";

expect.addSnapshotSerializer(createSerializer(emotion));

describe("DropdownMenuItem", () => {
  it("renders", () => {
    const component = shallow(
      <DropdownMenuItem index={0} listLength={1}>
        item content content
      </DropdownMenuItem>
    );
    expect(toJson(component)).toMatchSnapshot();
  });
  it("renders disabled", () => {
    const component = shallow(
      <DropdownMenuItem index={0} listLength={1} disabled={true}>
        item content content
      </DropdownMenuItem>
    );
    expect(toJson(component)).toMatchSnapshot();
  });
  it("renders active", () => {
    const component = shallow(
      <DropdownMenuItem index={0} listLength={1} isActive={true}>
        item content content
      </DropdownMenuItem>
    );
    expect(toJson(component)).toMatchSnapshot();
  });
  it("renders selected", () => {
    const component = shallow(
      <DropdownMenuItem index={0} listLength={1} isSelected={true}>
        item content content
      </DropdownMenuItem>
    );
    expect(toJson(component)).toMatchSnapshot();
  });
  it("renders active and selected", () => {
    const component = shallow(
      <DropdownMenuItem
        index={0}
        listLength={1}
        isSelected={true}
        isActive={true}
      >
        item content content
      </DropdownMenuItem>
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
