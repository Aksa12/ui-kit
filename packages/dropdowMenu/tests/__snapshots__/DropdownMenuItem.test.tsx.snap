// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownMenuItem renders 1`] = `
.emotion-0 {
  padding-left: 16px;
  padding-right: 16px;
  padding-bottom: 8px;
  padding-top: 8px;
  margin-top: 8px;
  margin-bottom: 8px;
}

<div
  className="emotion-0"
>
  item content content
</div>
`;

exports[`DropdownMenuItem renders active 1`] = `
.emotion-0 {
  padding-left: 16px;
  padding-right: 16px;
  padding-bottom: 8px;
  padding-top: 8px;
  background-color: #F7F8F9;
  margin-top: 8px;
  margin-bottom: 8px;
}

<div
  className="emotion-0"
>
  item content content
</div>
`;

exports[`DropdownMenuItem renders active and selected 1`] = `
.emotion-0 {
  padding-left: 16px;
  padding-right: 16px;
  padding-bottom: 8px;
  padding-top: 8px;
  background-color: #F7F8F9;
  background-color: #7D58FF;
  color: #FFF;
  background-color: #704FE5;
  margin-top: 8px;
  margin-bottom: 8px;
}

<div
  className="emotion-0"
>
  item content content
</div>
`;

exports[`DropdownMenuItem renders disabled 1`] = `
.emotion-0 {
  padding-left: 16px;
  padding-right: 16px;
  padding-bottom: 8px;
  padding-top: 8px;
  margin-top: 8px;
  margin-bottom: 8px;
}

<div
  className="emotion-0"
>
  item content content
</div>
`;

exports[`DropdownMenuItem renders selected 1`] = `
.emotion-0 {
  padding-left: 16px;
  padding-right: 16px;
  padding-bottom: 8px;
  padding-top: 8px;
  background-color: #7D58FF;
  color: #FFF;
  margin-top: 8px;
  margin-bottom: 8px;
}

<div
  className="emotion-0"
>
  item content content
</div>
`;
